tokio/src/stream/mod.rs

```diff
@@ -88,6 +94,82 @@ pub trait StreamExt: Stream {
     {
         Map::new(self, f)
     }
+
+    /// Filters the values produced by this stream according to the provided
+    /// predicate.
+    ///
+    /// As values of this stream are made available, the provided predicate `f`
+    /// will be run against them. If the predicate
+    /// resolves to `true`, then the stream will yield the value, but if the
+    /// predicate resolves to `false`, then the value
+    /// will be discarded and the next value will be produced.
```

Comment:
Do you think it's worth highlighting that this differs from `futures-rs`'s version, in that it takes a synchronous predicate?

Reply:
nah, i think that would add confusion.